File: Binary-Search/1646-kth-missing-positive-number/kth-missing-positive-number.cpp
class Solution {
public:
    int findKthPositive(vector<int>& nums, int k) {
        int left = 0, right = nums.size() - 1;
        
        while(left <= right){
            int mid = left + (right - left) / 2;
            int missing = nums[mid] - mid - 1;

            if(missing >= k) right = mid - 1;
            else left = mid + 1;

            cout << right << " " << left;
        }

        return left + k;
    }
};

File: Binary-Search/1646-kth-missing-positive-number/kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, nums: List[int], k: int) -> int:
        left, right = 0, len(nums) - 1
        
        while left <= right:
            mid = left + (right - left) // 2
            missing = nums[mid] - mid - 1
            
            if missing >= k:
                right = mid - 1
            else:
                left = mid + 1

            print(right, left)
        
        return left + k
